Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealityUtilitiesProvider.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.InteropServices;
+
+#if UNITY_WSA
+using UnityEngine.XR.WSA;
+#endif // UNITY_WSA
+
+namespace Microsoft.MixedReality.Toolkit.WindowsMixedReality
+{
+    public class WindowsMixedRealityUtilitiesProvider : IWindowsMixedRealityUtilitiesProvider
```

Comment:
Is the XR SDK version of this going to be named something else (i.e. XrSdkMixedRealityUtilitiesProvider)? If so, I wonder if the interface itself IWindowsMixedRealityUtilitiesProvider should be named something a bit more generic (i.e. IMixedRealityReferenceFrameProvider). As it's named it feels like there's mostly just this 1-1 mapping of IWindowsMixedRealityUtilitiesProvider ->WindowsMixedRealityUtilitiesProvider, but maybe seeing the other implementation of this would make it much clearer.

Reply:
Hmm...good point. I left the full `WindowsMixedReality` in there since it's WMR-specific, which would probably make the other provider something long like...`WindowsMixedRealityXRSDKUtilitiesProvider` or `XRSDKWindowsMixedRealityUtilitiesProvider`, which matches my naming so far but is a bit wordy.

I've been keeping scripts which use the existing XR pipeline with the same naming patterns we've been using (as opposed to adding `Legacy` or something) and adding `XRSDK` to differentiate.

Reply:
Along the lines of @wiwei's question; should we have an overall XR SDK utilities provider that aggregates anything we need to access, regardless of platform?  Is that feasible?

Reply:
We may move to a pattern like that in the future, but I've been thinking of these `Utilities` as platform-specific extensions, where we're able to implement additional platform-specific features based on Unity's native pointers. I'm only familiar with the native pointers for WMR, though other platforms may have them. When we want/need to take advantage of another platform's native pointer in the future, we can best figure out how to do that then. In my opinion, making it explicitly platform-specific is my preferred pattern, since there may not be a straight forward abstraction for each concept.